### ISSUE TITLE
Wrong grand_total in vendor/magento/module-quote/Model/Quote/Address/Total/Grand.php

### ISSUE THREAD

**Issue by lemaryn** [magento/magento2#26012]
Hello guys!

The same issue is https://github.com/magento/magento2/issues/6734 but you closed it.
### Preconditions (*)
Magento ver. 2.2.10, 2.4-develop / PHP Version 7.2.25

php function array_sum calculate wrong amount in 
vendor/magento/module-quote/Model/Quote/Address/Total/Grand.php
Line 26 $totals = array_sum($total->getAllTotalAmounts());
Line 27 $baseTotals = array_sum($total->getAllBaseTotalAmounts());

### Steps to reproduce (*)
For replicate it you can run this php script which return output 2.8421709430404E-14 but must be 0.

```
<?php 
$getAllTotalAmounts = [ 
"subtotal" => 108.29, "tax" => 0, 
"discount_tax_compensation" => 21.66, 
"weee" => 0, "shipping" => 0, 
"shipping_discount_tax_compensation" => 0, 
"discount" => -129.95, 
"extra_tax" => 0, 
"weee_tax" => 0 ]; 
echo array_sum($getAllTotalAmounts);
```
### Expected result (*)
for this code 
$totals = array_sum($total->getAllTotalAmounts());
$baseTotals = array_sum($total->getAllBaseTotalAmounts());

$totals and $baseTotals must be 0

### Actual result (*)
$totals and $baseTotals = 2.8421709430404E-14 

For fix it I replaced in vendor/magento/module-quote/Model/Quote/Address/Total/Grand.php

```
Line 26 $totals = array_sum($total->getAllTotalAmounts());
Line 27 $baseTotals = array_sum($total->getAllBaseTotalAmounts());
```

on 
```
        foreach ($total->getAllTotalAmounts() as $itemTotal) {
            $totals = round($totals, 2) + round($itemTotal, 2);
        }
        foreach ($total->getAllBaseTotalAmounts() as $baseTotal) {
            $baseTotals = round($baseTotals, 2) + round($baseTotal, 2);
        }
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29726` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by engcom-Charlie** [comment]
Hello @lemaryn 
Could you create a PR for this issue?

**Comment by lemaryn** [comment]
Hello engcom-Charlie  ! 
Yes I'm preparing it.

**Comment by engcom-Charlie** [comment]
@lemaryn so you can assign this ticket : )

**Comment by lemaryn** [comment]
@engcom-Charlie  Ha ha , was crazy issue for replicate , but thank you :)

**Comment by YuliyaLabudova** [comment]
Hi @engcom-Charlie, I can take the bug to me, but I don't have access to it. Could you, please, assign the ticket to me?

**Comment by lemaryn** [comment]
Hello guys!
As I see this issue was fixed in 2.4 
        $grandTotal = $this->priceRounder->roundPrice($total->getGrandTotal() + $totals, 4);
        $baseGrandTotal = $this->priceRounder->roundPrice($total->getBaseGrandTotal() + $baseTotals, 4);

It $this->priceRounder->roundPrice fixed issue. But in 2.2.10 issue is still replication . 

Regards,
Serhii

**Comment by engcom-Charlie** [comment]
@lemaryn so we can close this ticket. 
@YuliyaLabudova sorry, but this issue is not relevant now
